packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -1105,26 +1117,25 @@ export abstract class FluidDataStoreContext
 	}
 
 	/**
-	 * Summarizer client should not have local changes. These changes can become part of the summary and can break
+	 * Readonly client, including summarizer, should not have local changes. These changes can become part of the summary and can break
 	 * eventual consistency. For example, the next summary (say at ref seq# 100) may contain these changes whereas
 	 * other clients that are up-to-date till seq# 100 may not have them yet.
 	 */
-	protected identifyLocalChangeInSummarizer(eventName: string, type?: string): void {
-		if (
-			this.clientDetails.type !== summarizerClientType ||
-			this.localChangesTelemetryCount <= 0
-		) {
+	protected identifyLocalChangeIfReadonly(eventName: string, type?: string): void {
+		if (!this.isReadOnly() || this.localChangesTelemetryCount <= 0) {
```

Comment:
What does this do for interactive clients that are starting in read mode, expecting their first edit to flip over to write (presumably don't want to log for that case)?  Sorry I always get confused about how that state manifests in the various layers/properties.

Reply:
readonly and connection mode are totally separate, there is no overlap